File: Tree/102. Binary Tree Level Order Traversal.java
class Solution {
    ArrayList<List<Integer>> a = new  ArrayList<List<Integer>>();
    int level = 1;
    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root==null){
            return a;
        }
        if(a.size()<level){
            a.add(new ArrayList<Integer>());
        }
        a.get(level-1).add(root.val);
        level++;
        levelOrder(root.left);
        levelOrder(root.right);
        level--;

        return a;
    }

}

File: Tree/102. Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> a;
    int level = 1;
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root==nullptr){
            return a;
        }
        if(a.size()<level){
            a.push_back(vector<int>());
        }
        a[level-1].push_back(root->val);
        level++;
        levelOrder(root->left);
        levelOrder(root->right);
        level--;
        
        return a;
    }
};